packages/mobile/src/components/audio/Audio.tsx

```diff
@@ -323,37 +394,82 @@ export const Audio = () => {
       uri: m3u8
     }
   }
+
+  let nextSource
+  if (nextOfflineTrackUri) {
+    nextSource = {
+      type: TrackType.Default,
+      uri: nextOfflineTrackUri
+    }
+    // TODO: remove feature flag - https://github.com/AudiusProject/audius-client/pull/2147
+  } else if (isStreamMp3Enabled && nextStreamingUri) {
+    nextSource = {
+      type: TrackType.Default,
+      uri: nextStreamingUri
+    }
+  } else if (nextM3u8) {
+    nextSource = {
+      type: TrackType.HLS,
+      uri: nextM3u8
+    }
+  }
+
   const currentUriRef = useRef<string | null>(null)
   const isPodcastRef = useRef<boolean>(false)
 
-  const handleSourceChange = useCallback(
-    async (newSource: { uri: string; type?: TrackType }) => {
-      const newUri = newSource.uri
-      if (currentUriRef.current !== newUri) {
-        currentUriRef.current = newUri
-        const imageUrl = trackImageSource?.source[2].uri ?? DEFAULT_IMAGE_URL
-        await TrackPlayer.reset()
-        await TrackPlayer.add({
+  const handleSourceChange = useCallback(async () => {
+    const newUri = source.uri
+    if (currentUriRef.current !== newUri) {
+      currentUriRef.current = newUri
+      const imageUrl = trackImageSource?.source[2].uri ?? DEFAULT_IMAGE_URL
+      const nextImageUrl =
+        nextTrackImageSource?.source[2].uri ?? DEFAULT_IMAGE_URL
+
+      await TrackPlayer.reset()
+      // NOTE: Adding two tracks into the queue to make sure that android has a next button on the lock screen and notification controls
```

Comment:
Does this work hitting next multiple times from the lock screen? I assume it's calling handleSourceChange again in that case.

Reply:
yea it would update the source and call the handle func multiple times there 👍